src/extensions/default/DebugCommands/main.js

```diff
@@ -257,6 +273,485 @@ define(function (require, exports, module) {
+                // is defined. Check if that is valid or not.
+                prefType = _type;
+                if (!_isSupportedPrefType(prefType)) {
+                    prefType = "undefined";
+                }
+            } else if (Array.isArray(prefObj)) {
+                // Check if the object itself
+                // is an array, in which case
+                // we log the default.
+                prefType = "array";
+            } else if (prefObj.initial !== undefined  ||
+                       prefObj.keys !== undefined) {
+
+                // OK looks like this preference has
+                // no explicit type defined. instead
+                // it needs to be deduced from initial/keys
+                // variable.
+                var _prefVar;
+                if (prefObj.initial !== undefined) {
+                    _prefVar = prefObj.initial;
+                } else {
+                    _prefVar = prefObj.keys;
+                }
+
+                if (Array.isArray(_prefVar)) {
+                    // In cases of array the
+                    // typeof is returning a function.
+                    prefType = "array";
+                }
+
+            } else {
+                prefType = typeof (prefObj);
+            }
+        }
+
+        // Now make sure we recognize this format.
+        if (!_isSupportedPrefType(prefType)) {
+            prefType = "undefined";
+        }
+
+        return prefType;
+    }
+
+    function _isValidPref(pref) {
+
+        // Make sure to generate pref description only for
+        // user overrides and don't generate for properties
+        // meant to be used for internal purposes. Also check
+        // if the preference type is valid or not.
+        if (pref && !pref.excludeFromHints && _getObjType(pref) !== "undefined") {
+            return true;
+        }
+
+        return false;
+    }
+
+   /*
+    * This method tries to matach between initial objects
+    * and key objects and then aggregates objects from both
+    * the properties.
+    */
+    function _getObjKeys(prefObj) {
+
+        var finalObj = {},
+            property,
+            keysFound = false;
+
+        if (!prefObj) {
+            return {};
+        }
+
+        if (typeof (prefObj.initial) === "object") {
+            // iterate through the list.
+            keysFound = true;
+            for (property in prefObj.initial) {
+                if (prefObj.initial.hasOwnProperty(property)) {
+                    finalObj[property] = prefObj.initial[property];
+                }
+            }
+        }
+
+        if (typeof (prefObj.keys) === "object") {
+            // iterate through the list.
+            var allKeys = prefObj.keys;
+            keysFound = true;
+            for (property in allKeys) {
+                if (allKeys.hasOwnProperty(property)) {
+                    finalObj[property] = prefObj.keys[property];
+                }
+            }
+        }
```

Comment:
The common logic you use 3 times (that is: setting `keysFound = true`, looping over all keys and aggregating them) can be put into a function.